src/client/ProjectService.ts

```diff
@@ -4,21 +4,21 @@
 
 import { ClientServiceConfig } from './ClientServiceConfig'
 import { HDict, HStr } from 'haystack-core'
-import { HGrid, HaysonDict } from 'haystack-core'
+import { HGrid, HaysonDict, HMarker } from 'haystack-core'
 import { fetchVal } from './fetchVal'
 
 /**
  * A project record.
  */
 export interface Project extends HDict {
-	name?: HStr
-	meta?: HDict
+	proj: HMarker
+	projName: HStr
 }
 
 /**
  * An implementation of the FIN project service.
  */
-export class ProjectService {
+export class ProjectService<T extends Project = Project> {
```

Comment:
Prefer ProjectType to T. It's more readable.

Reply:
Updated. Also updated User, Roles and Groups to use the same naming structure rather than `T`